### PR TITLE
docs(addition): add docs for generalised restaking layer

### PR DESCRIPTION
**Pull request by JafarAz**
Required for merge:
- [ ] `pr-workflow-check / draft-release-check` is ✅ success
- Other rules GitHub shows you, or can be read in [configuration](github.com/ComposableFi/env/terraform/github.com/branches.tf) 

Makes review faster:
- [x] PR title is my best effort to provide summary of changes and has clear text to be part of release notes 
- [x] I marked PR by `misc` label if it should not be in release notes
- [ ] Linked Zenhub/Github/Slack/etc reference if one exists
- [ ] I was clear on what type of deployment required to release my changes (node, runtime, contract, indexer, on chain operation, frontend, infrastructure) if any in PR title or description
- [ ] Added reviewer into `Reviewers`
- [ ] I tagged(`@`) or used other form of notification of one person who I think can handle best review of this PR
- [ ] I have proved that PR has no general regressions of relevant features and processes required to release into production
- [ ] Any dependency updates made, was done according guides from relevant dependency
- Clicking all checkboxes 
- Adding detailed description of changes when it feels appropriate (for example when PR is big)

### CODE REVIEW

File: docs/docs/technology/restaking.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+:::tip
+The first opportunity for restaking is live on Solana. Users have a chance to get in on the action early and earn boosted rewards through [MANTIS games](../technology/restaking/mantis-games.md) - a team staking competition on [mantis.app](https://mantis.app/).
+:::
+
+Restaking has been pioneered and popularized by EigenLayer, which is a protocol for restaking ETH on Ethereum. In particular, users staking ETH are able to opt into EigenLayer’s smart contracts for restaking their ETH and thus extending the crypto economic security to additional applications within the ecosystem. EigenLayer thus addresses rising concerns of fragmented security on Ethereum, helping to bootstrap the security of various protocols/applications. EigenLayer’s total value locked (TVL) at the time of writing is over $275 million, indicating that there is a clear demand for restaking.
```
Comment (blasrodri) [inline docs/docs/technology/restaking.md:13]: isn't the TVL of EL way way higher than 275M?

Comment (0xBrainjar2) [reply docs/docs/technology/restaking.md:13]: indeed

File: docs/docs/technology/restaking/architecture.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+3. Monad LSTs and Native Withdrawal 
+4. TRX LSTs 
+5. lsDOT
+6. BNB LSTs 
+7. Cosmos assets via Picasso: 
+   - stTIA/milkTIA 
+   - stDYDX 
+   - stATOM
+   - SEI LSTs 
+   - qPICA
```
Comment (0xBrainjar2) [inline docs/docs/technology/restaking/architecture.md:23]: pica LSTs

**Review by 0xBrainjar2 — APPROVED** [review]

File: docs/docs/technology/restaking.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+:::
+
+Restaking has been pioneered and popularized by EigenLayer, which is a protocol for restaking ETH on Ethereum. In particular, users staking ETH are able to opt into EigenLayer’s smart contracts for restaking their ETH and thus extending the crypto economic security to additional applications within the ecosystem. EigenLayer thus addresses rising concerns of fragmented security on Ethereum, helping to bootstrap the security of various protocols/applications. EigenLayer’s total value locked (TVL) at the time of writing is over $7.5 billion, indicating that there is a clear demand for restaking.
+
+Despite the benefits of restaking, this concept has largely not yet expanded beyond the Ethereum ecosystem. However, there is a huge potential for restaking on other chains. Leveraging the fact that Picasso is a L1 Cosmos SDK chain that is also a hub for cross-ecosystem IBC, we are able to make this restaking layer cross-chain. Thus, vaults associated with this system will exist exclusively on IBC-enabled chains. Registration and accounting of AVSes are managed on Picasso. Vault contracts are to be deployed on Solana, Picasso and Ethereum in H1.
```
Comment (blasrodri) [inline docs/docs/technology/restaking.md:15]: maybe add a comment of what consequences this brings? More liquidity, but also more competitive yields for AVSes?

**Review by blasrodri — DISMISSED** [review]

File: docs/docs/technology/restaking/roadmap.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Roadmap
+
+The rollout plan for Picasso Generalized restaking will proceed as follows:
+
+1. Initial implementation of the [Restaking Vaults on Solana](../restaking/vaults.md).
+2. Launch of the first [AVS for Solana IBC](../restaking/sol-ibc-avs.md).
+3. Begin onboarding AVSs onto the Solana Restaking layer.
```
Comment (blasrodri) [inline docs/docs/technology/restaking/roadmap.md:7]: We need the generalized restaking layer to be available on the cosmos chain earlier

**Review by blasrodri — DISMISSED** [review]

**Review by blasrodri — APPROVED** [review]

**Pull request merged**